app/src/androidTest/java/com/github/se/cyrcle/ui/testEnd2End/MainActivityTest.kt

```diff
@@ -62,6 +89,85 @@ class MainActivityTest {
     cardRobot.assertParkingDetailsScreen()
   }
 
+  // ============================================================================
+  // ============================ ADDPARKINGS ROBOT =============================
+  // ============================================================================
+  private class AddParkingRobot(val composeTestRule: ComposeTestRule) {
+
+    fun assertLocationPickerScreen() {
+      composeTestRule.onNodeWithTag("LocationPickerScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("LocationPickerBottomBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("LocationPickerTopBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("cancelButton").assertIsDisplayed().assertHasClickAction()
+      composeTestRule.onNodeWithTag("nextButton").assertIsDisplayed().assertHasClickAction()
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun selectLocation() {
+      composeTestRule.onNodeWithTag("nextButton").assertHasClickAction().performClick()
+
+      composeTestRule.waitUntilAtLeastOneExists(hasTestTag("toggleRectangleButton"))
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun makeRectangleAndNext() = runTest {
+      composeTestRule.onNodeWithTag("LocationPickerScreen").performTouchInput {
+        swipe(start = Offset(0.5f, 0.5f), end = Offset(0.7f, 0.7f), durationMillis = 10)
+      }
+      composeTestRule.awaitIdle()
+      composeTestRule.onNodeWithTag("nextButton").performClick()
+
+      composeTestRule.waitUntilAtLeastOneExists(hasTestTag("AttributesPickerScreen"))
+    }
+
+    fun assertAttributesPickerScreen() {
+      composeTestRule.onNodeWithTag("AttributesPickerScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("AttributesPickerTopBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("AttributesPickerBottomBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("cancelButton").assertIsDisplayed().assertHasClickAction()
```

Comment:
The test tags are not consistent (some start with a Maj, some don't). Is it normal ?

Reply:
It's dependent on how people implemented their screens, a cleanup job will have to be done at some point for this...

Reply:
We will not forget to add that to the scrumboard as a future task.

Reply:
I still don't see a problem w/ the naming though. The first three are test tags for the screens hence they're in all caps. The last one's for a button so it's not. Anyways this isn't too big an issue but it may be a good idea to re-assess and re-verify the naming consistencies in our project as we head into M3 regardless